File: 디자인패턴/Python 디자인 패턴/Flyweight/DigitFactory.py
from Digit import*
class DigitFactory():
    pool = ["", "", "", "", "",
            "", "", "", "", ""]
    def getDigit(this, n:int)->Digit:
        if(this.pool[n] != ""):
            return this.pool[n]
        else:
            digit = Digit(f'./deta/{n}.txt')
            return digit.getData()
    pass

File: 디자인패턴/java 디자인 패턴/Flyweight/DigitFactory.java
public class DigitFactory {
    private Digit[] pool = new Digit[] {
        null, null, null, null, null, null,
        null, null, null, null, null, null
    };

    public Digit getDigit(int n) {
        if(pool[n] != null) {
            return pool[n];
        }
        else {
            String filename = String.format(".\\deta\\%d.txt", n);
            Digit digit = new Digit(filename);
            pool[n] = digit;
            return digit;
        }
    }
}
